src/test/java/io/aiven/elasticsearch/storage/io/CryptoIOProviderTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.elasticsearch.storage.io;
+
+import java.nio.channels.Channels;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.SecureRandom;
+import java.util.zip.CRC32;
+
+import io.aiven.elasticsearch.storage.RsaKeyAwareTest;
+import io.aiven.elasticsearch.storage.security.Encryption;
+import io.aiven.elasticsearch.storage.security.EncryptionKeyProvider;
+
+import com.google.common.io.ByteStreams;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CryptoIOProviderTest extends RsaKeyAwareTest implements Encryption {
+
+    private static final int BUFFER_SIZE = 8192;
+
+    private EncryptionKeyProvider encProvider;
+
+    private CryptoIOProvider cryptoIOProvider;
+
+    private SecureRandom secureRandom;
+
+    @BeforeEach
+    public void setUpKey() throws Exception {
+        encProvider =
+                EncryptionKeyProvider.of(
+                        Files.newInputStream(publicKeyPem),
+                        Files.newInputStream(privateKeyPem)
+                );
+        final var encryptionKey = encProvider.createKey();
+        cryptoIOProvider = new CryptoIOProvider(encryptionKey);
+        secureRandom = SecureRandom.getInstanceStrong();
```

Comment:
Maybe a nit, so feel free to ignore. But I'd switch to simple `Random` for tests.
Also, it can be just test-local variable.